core/match_data.rbs

```diff
@@ -45,6 +45,7 @@
 # See also "Special global variables" section in Regexp documentation.
 #
 class MatchData
+  type capture = String | Symbol | int
```

Comment:
Can they accept `_ToStr`?
```suggestion
  type capture = interned | int
```

Reply:
Sadly, no :-( It goes through `match_backref_number` which is (comments mine):
```c

static int
match_backref_number(VALUE match, VALUE backref)
{
    const char *name;
    int num;

    struct re_registers *regs = RMATCH_REGS(match);
    VALUE regexp = RMATCH(match)->regexp;

    match_check(match);
    if (SYMBOL_P(backref)) { // if backref is a `Symbol` (and not a subclass):
	backref = rb_sym2str(backref); // convert the symbol to a string
    }
    else if (!RB_TYPE_P(backref, T_STRING)) { // if backref is not a `String`
	return NUM2INT(backref); // call `.to_int` on it
    }
    name = StringValueCStr(backref); // convert the String to a null-terminated string

    num = name_to_backref_number(regs, regexp, name, name + RSTRING_LEN(backref));

    if (num < 1) {
        name_to_backref_error(backref);
    }

    return num;
}
```